src/App.tsx

```diff
@@ -14,11 +14,196 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const lengths: number[] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+interface State {
+  goods: string[];
+  isStarted: boolean;
+  isReversed: boolean;
+  sortBy: 'none' | 'name' | 'length';
+  selection: number;
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    isStarted: false,
+    isReversed: false,
+    sortBy: 'none',
+    selection: 1,
+  };
+
+  start = () => {
+    this.setState({ isStarted: true });
+  };
+
+  hide = () => (this.setState(state => (
+    {
+      isStarted: !state.isStarted,
+      isReversed: false,
+      sortBy: 'none',
+      selection: 1,
+    }
+  )));
```

Comment:
i might be wrong,
but it looks like you're not only hiding,
but rather `toggleVisibility`
`isStarted: !state.isStarted` - this swaps the state
```suggestion
  toggleVisibility = () => (this.setState(state => (
    {
      isStarted: !state.isStarted,
      isReversed: false,
      sortBy: 'none',
      selection: 1,
    }
  )));
```

Reply:
also you seem to also reset all the setting.
Could be generally, but you already have reset as a separate method.
So let's not repeat the functionality
```suggestion
  toggleVisibility = () => (this.setState(state => (
    {
      isStarted: !state.isStarted,
    }
  )));
```

Reply:
all done...